server/src/middleware/merchant-auth.ts

```diff
@@ -29,13 +36,16 @@ const merchantAuth = async (
       throw new Error('Missing Authorization header.');
 
     const [_, firebaseIdToken] = req.headers.authorization.split(' ');
-    const decodedFirebaseIdToken = await admin
-      .auth()
-      .verifyIdToken(firebaseIdToken);
-    req.decoded = decodedFirebaseIdToken;
+    const verifiedFirebaseUid = await getVerifiedUid(firebaseIdToken);
+
+    const {firebaseUid} = req.body;
+    if (firebaseUid !== undefined && firebaseUid !== verifiedFirebaseUid)
+      throw new Error('Invalid bearer token.');
+
     next();
-  } catch (error) {
-    res.sendStatus(401);
+  } catch (err) {
+    err.status = 401;
+    next(err);
```

Comment:
I'm curious on what the next function does? handle error?

Reply:
@xlai20 `next` is an express parameter. Basically for express there are a series of functions called middlewares, handling the `request` and `response` objects. Then, `next` just means like this function is done and now it's passing on the responsibility to the next function in line. Here, the next function after this is an error handler which then handles the error.

For better explanation you can read with example here: https://expressjs.com/en/guide/using-middleware.html